docs/apis/batch/libs/gelly.md

```diff
@@ -1734,3 +1734,547 @@ vertex represents a group of vertices and each edge represents a group of edges
+    xmlns:xlink="http://www.w3.org/1999/xlink">
+
+    <line x1="30" y1="40" x2="510" y2="40" />
+
+    <circle cx="30" cy="40" r="20" />
+    <text x="30" y="40">0</text>
+
+    <circle cx="150" cy="40" r="20" />
+    <text x="150" y="40">1</text>
+
+    <circle cx="270" cy="40" r="20" />
+    <text x="270" y="40">2</text>
+
+    <circle cx="390" cy="40" r="20" />
+    <text x="390" y="40">3</text>
+
+    <circle cx="510" cy="40" r="20" />
+    <text x="510" y="40">4</text>
+</svg>
+
+### RMat Graph
+
+A directed or undirected power-law graph generated using the [Recursive Matrix (R-Mat)]
+(http://www.cs.cmu.edu/~christos/PUBLICATIONS/siam04.pdf) model. RMat is a stochastic
+generator configured with a source of randomness.
+
+<div class="codetabs" markdown="1">
+<div data-lang="java" markdown="1">
+{% highlight java %}
+ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+
+RandomGenerableFactory<JDKRandomGenerator> rnd = new JDKRandomGeneratorFactory();
+
+int vertexCount = 1 << scale;
+int edgeCount = edgeFactor * vertexCount;
+
+Graph<LongValue,NullValue,NullValue> graph = new RMatGraph<>(env, rnd, vertexCount, edgeCount)
+    .generate();
+{% endhighlight %}
+</div>
+
+<div data-lang="scala" markdown="1">
+{% highlight scala %}
+import org.apache.flink.api.scala._
+import org.apache.flink.graph.generator.RMatGraph
+
+val env = ExecutionEnvironment.getExecutionEnvironment
+
+val vertexCount = 1 << scale
+val edgeCount = edgeFactor * vertexCount
+
+val graph = new RMatGraph(env.getJavaEnv, rnd, vertexCount, edgeCount).generate()
+{% endhighlight %}
+</div>
+</div>
+
+Manipulating the RMat constants and noise affects the degree-skew. The RMat generator can be
+configured to produce a simple graph by removing self-loops and duplicate edges, with undirected
+edges produced by a "clip-and-flip" throwing away the half matrix above the diagonal or a full
+"flip" preserving and mirroring all edges.
+
+<div class="codetabs" markdown="1">
+<div data-lang="java" markdown="1">
+{% highlight java %}
+ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+
+RandomGenerableFactory<JDKRandomGenerator> rnd = new JDKRandomGeneratorFactory();
+
+int vertexCount = 1 << scale;
+int edgeCount = edgeFactor * vertexCount;
+
+Graph<LongValue,NullValue,NullValue> graph = new RMatGraph<>(env, rnd, vertexCount, edgeCount)
+    .setConstants(0.57f, 0.19f, 0.19f)
+    .setNoise(true, 0.10f)
+    .setSimpleGraph(true, false)
+    .generate();
+{% endhighlight %}
+</div>
+
+<div data-lang="scala" markdown="1">
+{% highlight scala %}
+import org.apache.flink.api.scala._
+import org.apache.flink.graph.generator.RMatGraph
+
+val env = ExecutionEnvironment.getExecutionEnvironment
+
+val vertexCount = 1 << scale
+val edgeCount = edgeFactor * vertexCount
+
+val graph = new RMatGraph(env.getJavaEnv, rnd, vertexCount, edgeCount).setConstants(0.57f, 0.19f, 0.19f).setNoise(true, 0.10f).setSimpleGraph(true, false).generate()
```

Comment:
It would be helpful to add descriptions about `setConstants`, `setNoise` and other available configuration options.